warehouse/query-core/src/test/java/datawave/query/data/parsers/FieldIndexKeyTest.java

```diff
@@ -0,0 +1,69 @@
+package datawave.query.data.parsers;
+
+import org.apache.accumulo.core.data.Key;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class FieldIndexKeyTest {
+    
+    private final Key fiKey = new Key("row", "fi\0FIELD", "value\0datatype\0uid");
+    private final Key fiKeyWithNullsInValue = new Key("row", "fi\0FIELD", "v\0al\0ue\0datatype\0uid");
+    private final Key fiKeyWithChildUid = new Key("row", "fi\0FIELD", "value\0datatype\0uid.12.37");
```

Comment:
Consider adding some just plain wrong keys and testing error conditions.

Reply:
Added several error cases